server/container/guice/guice-common/src/main/java/org/apache/james/GuiceJamesServer.java

```diff
@@ -74,10 +74,18 @@ public void start() throws Exception {
         preDestroy = injector.getInstance(Key.get(new TypeLiteral<Stager<PreDestroy>>() {}));
         injector.getInstance(ConfigurationsPerformer.class).initModules();
         serverProbe = injector.getInstance(GuiceServerProbe.class);
-        jmapPort = injector.getInstance(JMAPServer.class).getPort();
+        jmapPort = locateJMAPPort(injector);
         webadminPort =locateWebAdminPort(injector);
     }
 
+    private Optional<Integer> locateJMAPPort(Injector injector) {
+        try {
+            return Optional.of(injector.getInstance(JMAPServer.class).getPort());
+        } catch(Exception e) {
```

Comment:
Basically all those called functions do not throw any exception but the catch exception should be there for any RuntimeException I think